src/index/Index.h

```diff
@@ -370,13 +372,14 @@ class Index {
    * Index class).
    */
   template <class Permutation>
-  void scan(Id key, IdTable* result, const Permutation& p) const {
+  void scan(Id key, IdTable* result, const Permutation& p,
+            ad_utility::SharedConcurrentTimeoutTimer timer = nullptr) const {
     if (p._meta.relationExists(key)) {
       const FullRelationMetaData& rmd = p._meta.getRmd(key)._rmdPairs;
       result->reserve(rmd.getNofElements() + 2);
       result->resize(rmd.getNofElements());
       p._file.read(result->data(), rmd.getNofElements() * 2 * sizeof(Id),
-                   rmd._startFullIndex);
+                   rmd._startFullIndex, std::move(timer));
```

Comment:
Ok, that gives the answer to my earlier question in IndexScan.cpp . I wonder though, why this function is here and not in IndexScan.cpp . Especially since it seems to be the only timeout-relevant function in class Index

Reply:
Yes, I came to the same conclusion above, but I think it is not relevant for the timeout PR